src/aspire/estimation/covar2d.py

```diff
@@ -9,25 +9,107 @@
 
 from aspire.utils.matrix import shrink_covar
 from aspire.utils.optimize import conj_grad
-from aspire.denoise.covar2d import RotCov2D
-from aspire.utils import ensure
 
+from aspire.utils import ensure
 
 logger = logging.getLogger(__name__)
 
 
-class Cov2DCTF(RotCov2D):
+class RotCov2D:
     """
-    Define a derived class for denoising 2D images using CTF information and the Covariance Wiener Filtering (CWF)
-    Cov2D method described in
+    Define a class for denoising 2D images using the Cov2D method with CTF information described in
 
     T. Bhamre, T. Zhang, and A. Singer, "Denoising and covariance estimation of single particle cryo-EM images",
     J. Struct. Biol. 195, 27-81 (2016). DOI: 10.1016/j.jsb.2016.04.013
     """
 
+    def __init__(self, basis, as_type='single'):
+        """
+        constructor of an object for 2D covariance analysis
+        """
+        self.basis = basis
+        ensure(basis.d == 2, 'Only two-dimensional basis functions are needed.')
+        self.as_type = as_type
+
+    def get_mean(self, coeffs):
+        """
+        Calculate the mean vector from the expansion coefficients of 2D images without CTF information.
+
+        :param coeffs: A coefficient vector (or an array of coefficient vectors) to be averaged.
+        :return: The mean value vector for all images.
+        """
+        if coeffs is None:
+            raise RuntimeError('The coefficients need to be calculated first!')
+
+        mask = self.basis._indices["ells"] == 0
+        mean_coeff = np.zeros((self.basis.basis_count, 1), dtype=self.as_type)
+        mean_coeff[mask, 0] = np.mean(coeffs[mask, ...], axis=1)
+
+        return mean_coeff
+
+    def get_covar(self, coeffs, mean_coeff=None,  do_refl=True):
+        """
+        Calculate the covariance matrix from the expansion coefficients without CTF information.
+
+        :param coeffs: A coefficient vector (or an array of coefficient vectors) calculated from 2D images.
+        :param mean_coeff: The mean vector calculated from the `coeffs`.
+        :param do_refl: If true, enforce invariance to reflection (default false).
+        :return: The covariance matrix of coefficients for all images.
+        """
+        if coeffs is None:
+            raise RuntimeError('The coefficients need to be calculated first!')
+        if mean_coeff is None:
```

Comment:
Same here.

Reply:
It allows to call the function directly for computing the covariance without calculating mean value first.

Reply:
Ok sure, but then a saner default would be to use a zero vector for the mean, I think.